File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int last = n+m-1;
        int i=m-1;
        int j=n-1;
        while (j>=0){
            if(i>=0 && nums1[i]>=nums2[j]){
                nums1[last] = nums1[i];
                i-=1;
            }else{
                nums1[last] = nums2[j];
                j-=1;
            }
            last-=1;
        }
    
    }
}

File: 0088-merge-sorted-array/0088-merge-sorted-array.kt
class Solution {
    fun merge(nums1: IntArray, m: Int, nums2: IntArray, n: Int): Unit {
        var last:Int = m+n-1
        var i = m-1
        var j=n-1;
        
        while(j>=0){
            if(i>=0 && nums1[i]>nums2[j]){
                nums1[last]=nums1[i]
                i--
            }else{
               nums1[last]=nums2[j]
                j--
            }
            last--
            
        }
        
    }
}